kong/db/schema/init.lua

```diff
@@ -1823,14 +1871,37 @@ function Schema.new_subschema(self, key, definition)
   assert(type(definition) == "table", "definition must be a table")
 
   if not self.subschema_key then
-    return nil, validation_errors.SUBSCHEMA_BAD_PARENT:format(self.name)
+    return nil, validation_errors.SUBSCHEMA_BAD_PARENT:format(key, self.name)
   end
 
   local subschema, err = Schema.new(definition, true)
   if not subschema then
     return nil, err
   end
 
+  local parent_by_name = {}
+  for _, f in ipairs(self.fields) do
+    local fname = next(f)
+    local fdata = f[fname]
```

Comment:
Should be `local fname, fdata = next(f)`

Reply:
ditto